lib/reply.js

```diff
@@ -157,16 +157,15 @@ function pumpCallback (reply) {
 
 function handleError (reply, error, cb) {
   var statusCode = reply.res.statusCode
-  if (error == null) {
-    statusCode = statusCode || 500
-  } else if (error.status >= 400) {
+
+  if (error.status >= 400) {
     statusCode = error.status
   } else if (error.statusCode >= 400) {
     statusCode = error.statusCode
-  } else {
-    statusCode = statusCode || 500
+  } else if (error && statusCode < 400) {
```

Comment:
Again, why `if (error)`?
Do we expect that `error` could not be an object and instead be `null` or `undefined`?
If so, this *if else* chain is wrong, because we will throw at the first *if*.

Reply:
When an error is passed with a status code < 400 `reply.code(200).send(new Error('foo'))` we have to overwrite it.

Reply:
Awesome, but why we need to check if `error` exist?
Few lines above you are doing `error.status`, this means that you know that `error` does exist.

If you know that `error` does exist then `if (error)` is useless.
If you know that `error` could not exist, you must add a `if (error == null)` check (double equal).

Reply:
You're right I was a bit lost I forgot that we handle only errros :D

Reply:
Fixed